lib/ae_mdw/sync/watcher.ex

```diff
@@ -1,64 +1,75 @@
 defmodule AeMdw.Sync.Watcher do
-  # credo:disable-for-this-file
-  use GenServer
+  @moduledoc """
+  Subscribes to chain events for dealing with new blocks and block
+  invalidations, and sends them to the syncing server.
+
+  There's two valid messages that arrive from core that we send to the
+  syncing server, plus 1 message that we handle internally:
+  * If there's a new key block added, we send the new height.
+  * If there's a new fork, we send `fork_height` to be the (height - 1)
+    that we need to revert to.
+  """
+
+  alias AeMdw.Sync.Server
 
-  import AeMdw.Util
+  use GenServer
 
-  defstruct [:operators, :sync_mon_ref]
+  defstruct []
 
-  ################################################################################
+  @typep t() :: %__MODULE__{}
 
-  def start_link(_),
+  @spec start_link(GenServer.options()) :: GenServer.on_start()
+  def start_link(_opts),
     do: GenServer.start_link(__MODULE__, [], name: __MODULE__)
 
-  def init([]) do
-    operators = Application.fetch_env!(:ae_mdw, :operators)
-    ref? = map_some(Process.whereis(AeMdw.Db.Sync), &Process.monitor/1)
-    {:ok, %__MODULE__{operators: operators, sync_mon_ref: ref?}}
-  end
+  @spec start_sync() :: :ok
+  def start_sync, do: GenServer.cast(__MODULE__, :start_sync)
+
+  @impl true
+  def init([]), do: {:ok, %__MODULE__{}}
 
-  def handle_cast({:sync_process, pid}, %__MODULE__{} = s),
-    do: {:noreply, %{s | sync_mon_ref: Process.monitor(pid)}}
+  @impl true
+  @spec handle_cast(:start_sync, t()) :: {:noreply, t()}
+  def handle_cast(:start_sync, state) do
+    :aec_events.subscribe(:chain)
+    :aec_events.subscribe(:top_changed)
 
-  def handle_info({:DOWN, ref, :process, _, :normal}, %__MODULE__{sync_mon_ref: ref} = s),
-    do: %{s | sync_mon_ref: nil}
+    Server.new_height(chain_height())
 
-  def handle_info({:DOWN, ref, :process, _, reason}, %__MODULE__{sync_mon_ref: ref} = s) do
-    notify_operators(s.operators, reason)
```

Comment:
I wonder if we could improve the notification feature rather than remove it. It might become helpful to be notified of unexpected behaviors with debugging info.

Reply:
I agree, but right now this functionality is of no use at all, writing files and sending emails with the `mail` command. I think we should work on an actual useful feature with a module called `ErrorNotifier` or some sort of more descriptive name

Reply:
okay